src/linagora.esn.calendar/app/calendar-configuration/calendar-configuration-tab-main/calendar-configuration-tab-main.controller.spec.js

```diff
@@ -469,6 +489,25 @@ describe('The calendar configuration tab delegation controller', function() {
       expect(CalendarConfigurationTabMainController.sharedCalendarOwner).to.equal(getOwnerResult);
       expect(CalendarConfigurationTabMainController.displayNameOfSharedCalendarOwner).to.equal(userUtilsResult);
     });
+
+    it('should create secret link with token to donwload ics calendar', function(done) {
+      const jwtPayload = {
+        calendarHomeId: 'calendarHomeId',
+        calendarId: 'calendarId',
+        userId: 'userid'
+      };
+
+      CalendarConfigurationTabMainController.$onInit();
+      $rootScope.$digest();
+      calendarService.generateTokenForSecretLink(jwtPayload)
```

Comment:
You are not supposed to be testing `calendarService.generateTokenForSecretLink` here, so you shouldn't call it here.